aioamqp/protocol.py

```diff
@@ -99,12 +99,13 @@ def close(self, no_wait=False, timeout=None):
         encoder.write_short(0)
         encoder.write_short(0)
         frame.write_frame(encoder)
-        if not no_wait:
-            yield from self.wait_closed(timeout=timeout)
+        yield from self.wait_closed(timeout=timeout)
 
     @asyncio.coroutine
     def wait_closed(self, timeout=None):
         yield from asyncio.wait_for(self.connection_closed.wait(), timeout=timeout, loop=self._loop)
+        for channel in tuple(self.channels.values()):
```

Comment:
The channels can change during iteration, so I think we do (test_channel_active_flow can be used to reproduce this one and test_connection_lost for the one on line 255).